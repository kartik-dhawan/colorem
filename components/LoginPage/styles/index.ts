export const styles = {
  loginPageWrapper: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    gap: "34px",
    border: "0.5px solid #727272",
    position: "absolute",
    top: "2rem",
    bottom: "2rem",
    left: "1.5rem",
    right: "1.5rem",
    padding: {
      xs: "0rem 2rem",
      md: "0rem 4rem",
    },
  },
  loginPageFormWrapper: {
    display: "flex",
    flexWrap: "wrap",
    gap: {
      xs: "16px",
      sm: "32px",
      lg: "48px",
    },
    alignItems: {
      xs: "center",
      sm: "flex-end",
    },
    flexDirection: {
      xs: "column",
      sm: "row",
    },
  },
  loginPageTextField: {
    width: {
      xs: "100%",
      sm: "40%",
      md: "25%",
      lg: "20%",
    },
    "& > .MuiFormLabel-root": {
      color: "#d9d9d9",
      fontWeight: 300,
      fontSize: "18px",
      "&.Mui-focussed": {
        color: "#727272 !important",
      },
      // "&.css-1c2i806-MuiFormLabel-root-MuiInputlabel-root.Mui-focused": {
      //   color: "#727272 !important",
      // },
    },
    "& > .MuiInputBase-root": {
      padding: "2px 8px",
      fontWeight: 300,
      fontSize: "18px",
      "&::before": {
        borderBottom: "0.5px solid #d9d9d9 !important",
      },
      "&::after": {
        borderBottom: "1px solid #d9d9d9 !important",
      },
      color: "#d9d9d9",
    },
  },
  loginPageButton: {
    height: "40px",
    color: "#222",
    padding: "16px 24px",
    width: {
      xs: "100%",
      sm: "40%",
      md: "30%",
    },
    backgroundColor: "#727272",
    borderRadius: "0px",
    marginTop: {
      xs: "1rem",
      sm: "0px",
    },
    "&:hover": {
      backgroundColor: "#b1b1b1",
    },
  },
  loginPageExtraOptions: {
    display: "flex",
    fontSize: "14px",
    justifyContent: {
      xs: "center",
      sm: "flex-start",
    },
  },
  loginPageExtraOptionsButton: {
    textTransform: "lowercase",
    fontWeight: 300,
    color: "#727272",
    transition: "500ms all ease",
    "&:hover": {
      backgroundColor: "transparent",
      color: "#b1b1b1",
    },
  },
}
