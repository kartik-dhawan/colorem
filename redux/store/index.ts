import { configureStore } from "@reduxjs/toolkit"
import countSlice from "../slices/countSlice"

const store = configureStore({
  reducer: {
    countSlice,
  },
})

export default store
