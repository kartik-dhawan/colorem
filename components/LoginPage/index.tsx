import { Box, Button, TextField } from "@mui/material"
import { styles } from "./styles"
import { useCallback, useEffect, useState } from "react"
import { useRouter } from "next/router"
import { LoginErrorSuccess, LoginFormState } from "../../utils/interfaces"
import { getAuth, signInWithEmailAndPassword } from "firebase/auth"
import db, { app } from "../../lib/auth/firebaseConfig"
import { collection, query, where } from "firebase/firestore"
import { getUsersDataFromQuery } from "../../lib/auth/firestore"
import LoadingButton from "@mui/lab/LoadingButton"
import { ErrorBoundary } from "react-error-boundary"
import ErrorFallback, { myErrorHandler } from "../common/ErrorFallback"
import { useDispatch, useSelector } from "react-redux"
import {
  updateAuthStatus,
  updateErrorSuccessState,
} from "../../redux/slices/authSlice"
import AuthAlert from "../common/AlertBoxes/AuthAlert"
import { RootType } from "../../redux/constants/stateTypes"
import { isValidEmail } from "../../utils/methods"

const LoginPage = () => {
  const lid = "loginPage"
  const router = useRouter()
  const auth = getAuth(app)
  const dispatch = useDispatch()
  const collectioRef = collection(db, "users")

  const initialFormState: LoginFormState = {
    email: "",
    username: "",
    password: "",
  }

  // finds the status of the page - login or signup
  const activityStatus = router.query?.activity
    ? router.query?.activity
    : "login"

  const [toggleLoginActivity, setToggleLoginActivity] = useState<boolean>(true) // true - login & false - sign up
  const [formData, setFormData] = useState<LoginFormState>(initialFormState)
  const [loader, setLoader] = useState<boolean>(false)
  const [disabledButton, setDisabledButton] = useState<boolean>(false) // eslint-disable-line

  const { errorSuccessState } = useSelector(
    (state: RootType) => state.authSlice
  )

  // logs user in on enter
  useEffect(() => {
    const keyDownHandler = (event: any) => {
      if (event.key === "Enter" && !disabledButton) {
        event.preventDefault()
        return loginHandler()
      }
    }

    document.addEventListener("keydown", keyDownHandler)

    return () => {
      document.removeEventListener("keydown", keyDownHandler)
    }
  }, [disabledButton, formData])

  /*
   * textfield input check to disable or enable buttons
   * also adds a check for password to be at least of 6 characters
   * & username to be of at least 4 characters
   */
  useEffect(() => {
    setDisabledButton(false)
    if (activityStatus === "login") {
      ;(formData.username.length < 4 || formData.password.length < 6) && // eslint-disable-line
        setDisabledButton(true)
    }
    if (activityStatus === "signup") {
      ;(formData.username.length < 4 || // eslint-disable-line
        formData.password.length < 6 ||
        !isValidEmail(formData.email)) &&
        setDisabledButton(true)
    }
  }, [formData])

  // enables url traversing to sign up or login page specificaly
  useEffect(() => {
    if (activityStatus === "login") {
      setToggleLoginActivity(true)
    } else if (activityStatus === "signup") {
      setToggleLoginActivity(false)
    }
  }, [router.query])

  const loginHandler = async () => {
    dispatch(updateErrorSuccessState({ status: null, error: null }))

    // initializes loader
    setLoader(true)

    // querying the firestore db to get email for the entered usernmame
    const q = query(collectioRef, where("username", "==", formData.username))
    console.log(formData)
    getUsersDataFromQuery(q).then((data: any) /* eslint-disable-line */ => {
      if (data !== null) {
        signInWithEmailAndPassword(auth, data.email, formData.password)
          .then(async (user) => {
            // algo : HS256
            const token = await user.user.getIdToken()
            localStorage.setItem("firebase-token", token)
            document.cookie = `firebase-token=${token}`
            // clears form data after login
            setFormData(initialFormState)
            // sets auth state in redux
            dispatch(updateAuthStatus(true))
            // post login redirect
            router.push("/dashboard")
          })
          .catch((error) => {
            // all errors will be handled in the same format
            const errorObject: LoginErrorSuccess = {
              status: "error",
              error: {
                code: error.code,
                message: error.message,
                name: error.name,
                customData: error.customData,
              },
            }
            dispatch(updateErrorSuccessState(errorObject))
            setLoader(false)
          })
      } else {
        const errorObject: LoginErrorSuccess = {
          status: "error",
          error: {
            name: "EmailNotFound",
            code: "auth/email-not-found",
            message:
              "The username is incorrect, the account does not exist. Please signup to continue.",
            customData: {},
          },
        }
        dispatch(updateErrorSuccessState(errorObject))
        setLoader(false)
      }
    })
  }

  const signUpHandler = () => {
    setLoader(true)
    router.push("/dashboard")
  }

  // toogles from signup page to login page
  const handleLoginToggle = useCallback(() => {
    setToggleLoginActivity(true)
    router.push({
      pathname: "/login",
      query: { ...router.query, activity: "login" },
    })
    setFormData(initialFormState)
  }, [router])

  // toogles from login page to signup page
  const handleSignupToggle = useCallback(() => {
    setToggleLoginActivity(false)
    router.push({
      pathname: "/login",
      query: { ...router.query, activity: "signup" },
    })
    setFormData(initialFormState)
  }, [router])

  return (
    <ErrorBoundary FallbackComponent={ErrorFallback} onError={myErrorHandler}>
      <Box
        className={lid + "FormWrapper"}
        id={lid + "FormWrapper"}
        data-testid={lid + "FormWrapper"}
        sx={styles.loginPageFormWrapper}
      >
        {errorSuccessState.status !== null && (
          <AuthAlert state={errorSuccessState} />
        )}
        {!toggleLoginActivity && (
          <TextField
            label="Email Id"
            type="email"
            autoComplete="current-password"
            variant="standard"
            className={lid + "TextField"}
            id={lid + "PasswordField"}
            data-testid={lid + "PasswordField"}
            required
            sx={{
              minWidth: {
                sm: "50%",
                md: "40%",
                lg: "35%",
              },
              ...styles.loginPageTextField,
            }}
            value={formData.email}
            onChange={(e) => {
              setFormData({ ...formData, email: e.target.value })
            }}
          />
        )}
        <TextField
          label="Username"
          variant="standard"
          className={lid + "TextField"}
          id={lid + "UsernameField"}
          data-testid={lid + "UsernameField"}
          sx={styles.loginPageTextField}
          value={formData.username}
          required
          onChange={(e) => {
            setFormData({ ...formData, username: e.target.value })
          }}
        />
        <TextField
          label="Password"
          type="password"
          autoComplete="current-password"
          variant="standard"
          className={lid + "TextField"}
          id={lid + "PasswordField"}
          data-testid={lid + "PasswordField"}
          required
          sx={styles.loginPageTextField}
          value={formData.password}
          onChange={(e) => {
            setFormData({ ...formData, password: e.target.value })
          }}
        />
        {toggleLoginActivity ? (
          <LoadingButton
            disableRipple
            sx={styles.loginPageButton}
            data-testid={lid + "LoginButton"}
            onClick={loginHandler}
            loading={loader}
            disabled={disabledButton}
          >
            Login
          </LoadingButton>
        ) : (
          <LoadingButton
            disableRipple
            sx={styles.loginPageButton}
            data-testid={lid + "SignupButton"}
            onClick={signUpHandler}
            loading={loader}
            disabled={disabledButton}
          >
            Sign Up
          </LoadingButton>
        )}
      </Box>
      <Box sx={styles.loginPageExtraOptions}>
        {toggleLoginActivity ? (
          <Button
            disableRipple
            sx={styles.loginPageExtraOptionsButton}
            onClick={handleSignupToggle}
            data-testid={lid + "ToggleSignup"}
          >
            create an account
          </Button>
        ) : (
          <Button
            disableRipple
            sx={styles.loginPageExtraOptionsButton}
            onClick={handleLoginToggle}
            data-testid={lid + "ToggleLogin"}
          >
            log into an account
          </Button>
        )}
        <Button disableRipple sx={styles.loginPageExtraOptionsButton}>
          recover account
        </Button>
        <Button disableRipple sx={styles.loginPageExtraOptionsButton}>
          log in with google
        </Button>
      </Box>
    </ErrorBoundary>
  )
}
export default LoginPage
