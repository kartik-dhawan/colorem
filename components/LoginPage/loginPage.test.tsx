import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import store from "../../redux/store"
import LoginPage from "."
import mockRouter from "next-router-mock"

jest.mock("next/router", () => require("next-router-mock"))

describe("Login page", () => {
  it("-- (login) should match snapshot", () => {
    mockRouter.push({
      pathname: "/login",
      query: { ...mockRouter.query, activity: "login" },
    })

    const tree = renderer
      .create(
        <Provider store={store}>
          <LoginPage />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
    expect(mockRouter).toMatchObject({
      query: {
        activity: "login",
      },
    })
  })

  it("-- (signup) should match snapshot", () => {
    mockRouter.push({
      pathname: "/signup",
      query: { ...mockRouter.query, activity: "signup" },
    })

    const tree = renderer
      .create(
        <Provider store={store}>
          <LoginPage />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
    expect(mockRouter).toMatchObject({
      query: {
        activity: "signup",
      },
    })
  })
})
