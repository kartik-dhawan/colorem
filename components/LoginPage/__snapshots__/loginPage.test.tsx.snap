// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component -- (login) should match snapshot 1`] = `
[
  <div
    className="loginPageFormWrapper MuiBox-root css-18l78r8"
    data-testid="loginPageFormWrapper"
    id="loginPageFormWrapper"
  >
    <div
      className="MuiFormControl-root MuiTextField-root loginPageTextField css-1nbje9m-MuiFormControl-root-MuiTextField-root"
      data-testid="loginPageUsernameField"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="loginPageUsernameField"
        id="loginPageUsernameField-label"
      >
        Username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={false}
          id="loginPageUsernameField"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root loginPageTextField css-1nbje9m-MuiFormControl-root-MuiTextField-root"
      data-testid="loginPagePasswordField"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="loginPagePasswordField"
        id="loginPagePasswordField-label"
      >
        Password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="current-password"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={false}
          id="loginPagePasswordField"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-11wldcc-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      data-testid="loginPageLoginButton"
      disabled={false}
      id=":r2:"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Login
    </button>
  </div>,
  <div
    className="MuiBox-root css-98mxar"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1l03z47-MuiButtonBase-root-MuiButton-root"
      data-testid="loginPageToggleSignup"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      create an account
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1l03z47-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      recover account
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1l03z47-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      log in with google
    </button>
  </div>,
]
`;

exports[`Login component -- (signup) should match snapshot 1`] = `
[
  <div
    className="loginPageFormWrapper MuiBox-root css-18l78r8"
    data-testid="loginPageFormWrapper"
    id="loginPageFormWrapper"
  >
    <div
      className="MuiFormControl-root MuiTextField-root loginPageTextField css-1wyy5f0-MuiFormControl-root-MuiTextField-root"
      data-testid="loginPagePasswordField"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="loginPagePasswordField"
        id="loginPagePasswordField-label"
      >
        Email Id
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="current-password"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={false}
          id="loginPagePasswordField"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root loginPageTextField css-1nbje9m-MuiFormControl-root-MuiTextField-root"
      data-testid="loginPageUsernameField"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="loginPageUsernameField"
        id="loginPageUsernameField-label"
      >
        Username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={false}
          id="loginPageUsernameField"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root loginPageTextField css-1nbje9m-MuiFormControl-root-MuiTextField-root"
      data-testid="loginPagePasswordField"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="loginPagePasswordField"
        id="loginPagePasswordField-label"
      >
        Password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="current-password"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={false}
          id="loginPagePasswordField"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-11wldcc-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      data-testid="loginPageSignupButton"
      disabled={true}
      id=":r5:"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      Sign Up
    </button>
  </div>,
  <div
    className="MuiBox-root css-98mxar"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1l03z47-MuiButtonBase-root-MuiButton-root"
      data-testid="loginPageToggleLogin"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      log into an account
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1l03z47-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      recover account
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1l03z47-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      log in with google
    </button>
  </div>,
]
`;
